app/assets/javascripts/legacy/views/question_form.js

```diff
@@ -106,16 +106,23 @@
     var question_type_param = '?adding_to_question_type=' + self.field_value('qtype_name'),
         loadUrl = self.params.new_option_set_path + question_type_param;
 
+    var modal_header = 'option_set.create_for_question';
+    var question_code = $('#question_code').val().trim();
+    if (question_code === '') {
+      modal_header = 'option_set.create_for_new_question';
+    }
+    $('#create-option-set .modal-title').html(I18n.t(modal_header, {code: question_code}));
```

Comment:
One thing: can you please change to `.text` instead of `.html` so that we get XSS protection.